### ISSUE TITLE
Implement Validator Rotation and Minimal Slashing

### ISSUE THREAD

**Issue by terenc3t** [prysmaticlabs/prysm#598]
@djrtwo introduced [validator rotation and minimal slashing](https://github.com/ethereum/eth2.0-specs/pull/17) for 2.1 today. This issue will address those functionalities within our `casper` packages. It includes:
- Handle special messages validators can sign outside of normal consensus
- Implement routine to add validator into the pool
- Track special messages in `SpecialObjects`
- Process these special messages during crystallized state transition
- Implement validator shuffling in and out of active set, also perform withdraws after withdrawal period has ended

I'll work on this after v0.0.0 completes

**Comment by djrtwo** [comment]
There are likely some edge cases that we haven't yet addressed in all this subtle logic.
I'll be implementing this week, but if you run in to anything that doesn't seem covered, please please let me know

**Comment by nisdas** [comment]
This is blocked by both #575 and #538

**Comment by terenc3t** [comment]
@nisdas Wanna split this up in two? You can take care of everything relates to rewards/penalties/slashing. I can take care of validators induction and the new `special object` field

**Comment by nisdas** [comment]
@terenc3t sounds good ! Before we get started on this lets resolve #575 and #538 first.

**Comment by terenc3t** [comment]
I'm starting new protobuf types and active state logic. Unrelated to those 2 :)

**Comment by nisdas** [comment]
Alright sounds good to me shouldn't have any problem getting those merged in first

**Comment by rauljordan** [comment]
can this be closed @terenc3t?

**Comment by terenc3t** [comment]
We haven't implemented all the slashing conditions. Since researching is still going on. I'm marking this blocked by research for now

**Comment by rauljordan** [comment]
This is closed by #1086 and #1076 along with the greater effort of #781
